data-prepper-plugins/geoip-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/geoip/provider/LocationData.java

```diff
@@ -1,43 +1,164 @@
-
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         LocationData that = (LocationData) o;
-        return Objects.equals(countryName, that.countryName) && Objects.equals(subdivisionName, that.subdivisionName) && Objects.equals(cityName, that.cityName);
+        return Objects.equals(ip, that.ip) && Objects.equals(countryName, that.countryName) && Objects.equals(regionName, that.regionName) && Objects.equals(cityName, that.cityName) && Objects.equals(location, that.location) && Objects.equals(latitude, that.latitude) && Objects.equals(longitude, that.longitude) && Objects.equals(continentCode, that.continentCode) && Objects.equals(countryCode, that.countryCode) && Objects.equals(postalCode, that.postalCode) && Objects.equals(regionCode, that.regionCode) && Objects.equals(timeZone, that.timeZone);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(countryName, subdivisionName, cityName);
+        return Objects.hash(ip, countryName, regionName, cityName, location, latitude, longitude, continentCode, countryCode, postalCode, regionCode, timeZone);
+    }
+
+    public static final class Builder {
+        private String ip;
+        private String countryName;
+        private String regionName;
+        private String cityName;
+        private Double[] location;
+        private Double latitude;
+        private Double longitude;
+        private String continentCode;
+        private String countryCode;
+        private String postalCode;
+        private String regionCode;
+        private String timeZone;
+
+        public Builder withIp(String ip) {
+            this.ip = ip;
+            return this;
+        }
+
+        public Builder withCity(String city) {
+            this.cityName = city;
+            return this;
+        }
+
+        public Builder withCountry(String country) {
+            this.countryName = country;
+            return this;
+        }
+
+        public Builder withContinent(String continent) {
+            this.continentCode = continent;
+            return this;
+        }
+
+        public Builder withCountryCode(String countryCode) {
+            this.countryCode = countryCode;
+            return this;
+        }
+
+        public Builder withPostal(String postal) {
+            this.postalCode = postal;
+            return this;
+        }
+
+        public Builder withRegion(String region) {
+            this.regionName = region;
+            return this;
+        }
+
+        public Builder withRegionCode(String regionCode) {
+            this.regionCode = regionCode;
+            return this;
+        }
+
+        public Builder withTimeZone(String timeZone) {
+            this.timeZone = timeZone;
+            return this;
+        }
+
+        public Builder withLocation(Double[] location) {
+            if (location.length != 2)
```

Comment:
nit - prefer braces always, less jarring to read and easier to modify in the future